File: food_project_mobile/lib/models/produk_model.dart
class ProdukModel {
  final int id;
  final String nama;
  final int harga;
  final List<String> gambar;
  final String deskripsi;
  final int idKategori;

  ProdukModel({
    required this.id,
    required this.nama,
    required this.harga,
    required this.gambar,
    required this.deskripsi,
    required this.idKategori,
  });
}

File: food_project_admin/server/models/produk_model.js
const { DataTypes } = require("sequelize")
const sequelize = require("../config/database")
const KategoriModel = require("./kategori_model")
const GambarProdukModel = require("./gambar_produk_model")

const ProdukModel = sequelize.define("produk", {
  id: {
    type: DataTypes.INTEGER,
    allowNull: false,
    primaryKey: true,
    autoIncrement: true,
  },
  nama: {
    type: DataTypes.STRING,
    allowNull: false,
  },
  deskripsi: {
    type: DataTypes.TEXT,
    allowNull: false,
  },
  harga: {
    type: DataTypes.INTEGER,
    allowNull: false,
  },
  idKategori: {
    type: DataTypes.INTEGER,
    allowNull: false,
  },
  stock: {
    type: DataTypes.INTEGER,
    allowNull: false,
  },
}, {
  freezeTableName: true,
})

ProdukModel.belongsTo(KategoriModel, { foreignKey: "idKategori", onDelete: "CASCADE" })
ProdukModel.hasMany(GambarProdukModel, {
  as: "gambar",
  foreignKey: "idProduk",
  onDelete: "CASCADE"
})

module.exports = ProdukModel